axiom/nr_log_event.c

```diff
@@ -37,11 +37,83 @@ void nr_log_event_destroy(nr_log_event_t** ptr) {
   nr_realfree((void**)ptr);
 }
 
+nr_log_event_t* nr_log_event_clone(const nr_log_event_t* src) {
+  nr_log_event_t* clone = NULL;
+
+  if (NULL == src) {
+    return NULL;
+  }
+
+  clone = nr_log_event_create();
+  if (NULL == clone) {
+    return NULL;
+  }
+
+  if (NULL != src->trace_id) {
+    clone->trace_id = nr_strdup(src->trace_id);
+  }
+  if (NULL != src->log_level) {
+    clone->log_level = nr_strdup(src->log_level);
+  }
+  if (NULL != src->message) {
+    clone->message = nr_strdup(src->message);
+  }
+  if (NULL != src->span_id) {
+    clone->span_id = nr_strdup(src->span_id);
+  }
+  if (NULL != src->entity_guid) {
+    clone->entity_guid = nr_strdup(src->entity_guid);
+  }
+  if (NULL != src->entity_name) {
+    clone->entity_name = nr_strdup(src->entity_name);
+  }
+  if (NULL != src->hostname) {
+    clone->hostname = nr_strdup(src->hostname);
+  }
+  clone->priority = src->priority;
+  clone->timestamp = src->timestamp;
+
+  return clone;
+}
+
+static bool add_log_field_to_buf(nrbuf_t* buf,
+                                 const char* field_name,
+                                 const char* field_value,
+                                 const bool first,
+                                 const bool required) {
+  const char* final_value = field_value;
+
+  if (NULL == buf || nr_strempty(field_name)) {
+    return false;
+  }
+
+  if (nr_strempty(field_value)) {
+    if (!required) {
+      return false;
+    } else {
+      final_value = "null";
+    }
+  }
+
+  if (!first) {
+    nr_buffer_add(buf, NR_PSTR(","));
+  }
+  nr_buffer_add(buf, NR_PSTR("\""));
+  nr_buffer_add(buf, field_name, nr_strlen(field_name));
+  nr_buffer_add(buf, NR_PSTR("\""));
+  nr_buffer_add(buf, NR_PSTR(":"));
+  nr_buffer_add(buf, NR_PSTR("\""));
+  nr_buffer_add(buf, final_value, nr_strlen(final_value));
+  nr_buffer_add(buf, NR_PSTR("\""));
+
+  return true;
+}
+
 char* nr_log_event_to_json(const nr_log_event_t* event) {
   nrbuf_t* buf = nr_buffer_create(0, 0);
   char* json = NULL;
 
```

Comment:
check for NULL event?

Reply:
This is checked in the function which is used to actually format the json (`nr_log_event_to_json_buffer_ex()`) but I will add a check here see it is more obvious it is handled.

Fixed in commit [741a19b](https://github.com/newrelic/newrelic-php-agent/pull/515/commits/741a19b0bfc79066d8ffad066cc30a58238ee90b).